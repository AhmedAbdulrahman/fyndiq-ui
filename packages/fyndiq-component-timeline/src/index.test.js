import React from 'react'
import { shallow } from 'enzyme'

import { Timeline, Event } from './'

// eslint-disable-next-line
const Icon = ({ color }) => <div style={{ color }}>My Icon</div>

describe('fyndiq-component-timeline', () => {
  describe('Timeline Component', () => {
    it('should have the proper structure', () => {
      expect(
        shallow(
          <Timeline>
            <div>Hello</div>
          </Timeline>,
        ),
      ).toMatchSnapshot()
    })

    it('should have a type prop', () => {
      expect(
        shallow(<Timeline type="black" />).hasClass('timeline_type_black'),
      ).toBe(true)
    })
  })

  describe('Event Component', () => {
    it('should have the proper structure', () => {
      expect(shallow(<Event icon={<Icon />}>Hello</Event>)).toMatchSnapshot()
    })
  })
})
