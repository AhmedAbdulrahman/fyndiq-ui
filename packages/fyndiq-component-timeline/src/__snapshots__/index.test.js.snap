// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-timeline Event Component should have the proper structure 1`] = `
<div
  className="event"
>
  <div
    className="icon"
  >
    <Icon
      color={null}
    />
  </div>
  <span>
    Hello
  </span>
</div>
`;

exports[`fyndiq-component-timeline Timeline Component should have the proper structure 1`] = `
<div
  className="
      timeline
      timeline_type_white
    "
>
  <div>
    Hello
  </div>
</div>
`;
